### PR TITLE
[qemu] ensure `dnsmasq.hosts` exists

### PR DESCRIPTION
**Pull request by surahman**
**_With respect to issue #1118:_**
Added a check on `DNSMasq` bootstrapping for `dnsmasq.hosts` file in the constructor. If absent, an empty file will be created in the `data_dir`.

### CODE REVIEW

File: src/platform/backends/qemu/dnsmasq_server.cpp [hunk 1]
```diff
@@ -52,6 +52,13 @@ mp::DNSMasqServer::DNSMasqServer(const Path& data_dir, const QString& bridge_nam
     conf_file.open();
     conf_file.close();
 
+    QFile fd_Hosts(QDir(data_dir).filePath("dnsmasq.hosts"));
```
Comment (townsend2010) [inline src/platform/backends/qemu/dnsmasq_server.cpp:55]: Could you change the variable name to `fd_hosts` instead to match the naming scheme of other variables?

File: src/platform/backends/qemu/dnsmasq_server.cpp [hunk 2]
```diff
@@ -52,6 +52,13 @@ mp::DNSMasqServer::DNSMasqServer(const Path& data_dir, const QString& bridge_nam
...
+    if (!fd_Hosts.exists())
+    {
+        fd_Hosts.open(QIODevice::WriteOnly);
+        fd_Hosts.close();
```
Comment (townsend2010) [inline src/platform/backends/qemu/dnsmasq_server.cpp:59]: This `close()` isn't necessary as [`QFile`'s dtor](https://doc.qt.io/qt-5/qfile.html#dtor.QFile) will close it automatically, but it's not wrong to have it either.

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Hi @surahman!

Thanks for this!  I have a couple of inline niggles and I would also like to ask you to add a test under `tests/qemu/test_dnsmasq_server.cpp` please.

**Comment by Saviq** [comment]
Will reopen to kick the CLA check.

**Comment by surahman** [comment]
@Saviq sorry about, privacy setting on Github were complaining about email address exposure before.

**Comment by Saviq** [comment]
@surahman the `gitignore` commit still has the `-` email. But please remove that commit anyway :)

**Comment by Saviq** [comment]
> @Saviq sorry I thought I already had - I just filled it out again.

Thanks, maybe it didn't go through our legal wheels… Now I know you did, I'll follow up our side.

Thanks a bunch, again, for your help!

**Comment by townsend2010** [comment]
bors try

**Comment by townsend2010** [comment]
Ok, all we're missing a simple little test to ensure the `dnsmasq.hosts` file does indeed exist.  I see the codecov reports the line being covered, but I think an explicit test would be beneficial.

**Comment by surahman** [comment]
I have added a quick test. Is there anything I need to do to fix the CI issue?
`mkdir: cannot create directory '/etc/systemd/system': File exists`

**Comment by townsend2010** [comment]
Thanks!
> Is there anything I need to do to fix the CI issue?

I'll for this CI run to complete and then look at the error, if there is one.

**Comment by townsend2010** [comment]
Looks like CI passed fine with your added test.  I'm not sure where you saw that failure before though.

**Review by townsend2010 — APPROVED** [review]
Ok, this looks good.  Thanks!

bors merge

**Comment by Saviq** [comment]
Sorry @surahman, we're having some reliability issues with a couple tests in CI…

bors retry

**Comment by surahman** [comment]
@Saviq no worries, it is interesting to see the workflow :)

**Comment by Saviq** [comment]
bors retry

**Pull request merged**